### ISSUE TITLE
Unable to reproduce performance

### ISSUE THREAD

**Issue by guozhiyao** [uclaml/SPIN#12]
- base model: 'alignment-handbook/zephyr-7b-sft-full'
- train data: 'UCLA-AGI/SPIN_iter0'
I use the default hyper-parameter to train the model, and test the model with `HuggingFaceH4/open_llm_leaderboard` locally. The result on `allenai/ai2_arc` as below:
- base model: 0.5819112627986348
- epoch1 (step 778): 0.5989761092150171
- epoch2 (step 1556): 0.5964163822525598
- epoch3 (step 2334): 0.590443686006826
which can not match the performance in paper (63.40).

**Comment by angelahzyuan** [comment]
Try setting --num_train_epochs=6. We've included this as a comment and uncommented it in the revised finetune.sh script. You can still stop at the first few epochs.

**Comment by guozhiyao** [comment]
> Try setting --num_train_epochs=6. We've uncommented it in the revised finetune.sh script (previously specified in comments). You can still stop at the first few epochs.

@angelahzyuan Thanks for your reply, but the performance of the baseline model is not consistent with the results in the paper.
<img width="1785" alt="image" src="https://github.com/uclaml/SPIN/assets/21999339/22eb1019-d0be-48de-bd8e-4128c4d715bb">

**Comment by guozhiyao** [comment]
> Try setting --num_train_epochs=6. We've uncommented it in the revised finetune.sh script (previously specified in comments). You can still stop at the first few epochs.

@angelahzyuan I have change the epoch to 6. I test the model of epoch 1 on `allenai/ai2_arc`, which still can not match the result.
<img width="501" alt="image" src="https://github.com/uclaml/SPIN/assets/21999339/df95c2d0-763c-4bb7-89d2-f21a51726dd5">

**Comment by yihedeng9** [comment]
Please check our readme for the following,

1. We noticed that [Alignment Handbook](https://github.com/huggingface/alignment-handbook) has updated their configuration and SFT checkpoint since our experiments. The configuration and SFT model from the Alignment Handbook that we used in our experiments for data generation and fine-tuning are the older version ([Config](https://github.com/huggingface/alignment-handbook/blob/61a11a5c7d66179ed0a930b0dd12e532fce701dd/recipes/zephyr-7b-beta/dpo/config_full.yaml), [Model](https://huggingface.co/alignment-handbook/zephyr-7b-sft-full/tree/ac6e600eefcce74f5e8bae1035d4f66019e93190)). If you wish to use the newest SFT model, you need to generate your own data instead of using the [datasets](https://huggingface.co/collections/UCLA-AGI/datasets-spin-65c3624e98d4b589bbc76f3a) we provided on huggingface.

2. For our evaluation on the Open LLM Leaderboard, please use the [lm-evaluation-harness](https://github.com/EleutherAI/lm-evaluation-harness) repository at v0.4.0. Also, note that we set the number of few shot examples to be the same as instructed on the [Leaderboard](https://huggingface.co/spaces/HuggingFaceH4/open_llm_leaderboard), as in their 'About'. The leaderboard uses a different [version](https://github.com/EleutherAI/lm-evaluation-harness/tree/b281b0921b636bc36ad05c0b0b0763bd6dd43463) of lm-eval-harness. Different evaluation versions results in different scores, but the trend for improvement will remain the same. Lastly, we have also uploaded our models to the leaderboard, for which you can directly search for the results (https://huggingface.co/datasets/open-llm-leaderboard/details_UCLA-AGI__zephyr-7b-sft-full-SPIN-iter0).

**Comment by angelahzyuan** [comment]
> > Try setting --num_train_epochs=6. We've uncommented it in the revised finetune.sh script (previously specified in comments). You can still stop at the first few epochs.
> 
> @angelahzyuan I have change the epoch to 6. I test the model of epoch 1 on `allenai/ai2_arc`, which still can not match the result. <img alt="image" width="501" src="https://private-user-images.githubusercontent.com/21999339/306994531-df95c2d0-763c-4bb7-89d2-f21a51726dd5.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDg2MzU0OTAsIm5iZiI6MTcwODYzNTE5MCwicGF0aCI6Ii8yMTk5OTMzOS8zMDY5OTQ1MzEtZGY5NWMyZDAtNzYzYy00YmI3LTg5ZDItZjIxYTUxNzI2ZGQ1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAyMjIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMjIyVDIwNTMxMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTY4MjlmOGZiZTdhMWU4NTgyMmFhMjFlZTdhYTBhYzJlM2Q1YjVhMGQ4MjQxYTY0YTIyNGYwN2RmNDAzY2IxOTcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.fSCayd_alqGpe7jpufsSYV3VV8X2Lse18gH_jfGiUUY">

@guozhiyao Furthermore, we've retrained the Zephyr-7b-sft-full model with the latest updates from the [Alignment Handbook](https://github.com/huggingface/alignment-handbook). Our evaluation on ARC indicates an improvement from 57.51 (sft-full) to 60.75 (SPIN-iter0). While different base models and evaluation methods may produce varying results, the overall performance trend remains consistent.

**Comment by lewtun** [comment]
Hello, @guozhiyao I'm one of the authors of the `alignment-handbook` 👋 

Indeed, we updated some of the configs used to train the SFT model in this PR (https://github.com/huggingface/alignment-handbook/pull/88) because there were some bugs fixed in TRL that changes the learning rate scheduler.

If you want to use the original checkpoint we released with the handbook you can load the model with `revision=ac6e600eefcce74f5e8bae1035d4f66019e93190`

**Comment by lewtun** [comment]
One related question for @angelahzyuan - do you happen to know why the GSM8k values on the leaderboard are so different from those shown in your paper?

It seems like `iter0` is actually the best on the leaderboard, while your paper shows `iter2` and `iter3` are considerably better. 

![Screenshot 2024-02-23 at 10 13 28](https://github.com/uclaml/SPIN/assets/26859204/b9edf99d-ce1b-4989-9565-a08b4573c62a)

![Screenshot 2024-02-23 at 10 15 09](https://github.com/uclaml/SPIN/assets/26859204/e2cd5bee-c533-4df5-be22-d0395e32f0d9)

**Comment by yihedeng9** [comment]
Hi, the difference in scores between leaderboard and our papers is mainly due to difference in [lm-evaluation-harness](https://github.com/EleutherAI/lm-evaluation-harness) version. We used v.0.4.0, which has a different evaluation result as compared to the older [version](https://github.com/EleutherAI/lm-evaluation-harness/tree/b281b0921b636bc36ad05c0b0b0763bd6dd43463) used by leaderboard, especially in gsm8k as we observed.

**Comment by lewtun** [comment]
> Hi, the difference in scores between leaderboard and our papers is mainly due to difference in [lm-evaluation-harness](https://github.com/EleutherAI/lm-evaluation-harness) version. We used v.0.4.0, which has a different evaluation result as compared to the older [version](https://github.com/EleutherAI/lm-evaluation-harness/tree/b281b0921b636bc36ad05c0b0b0763bd6dd43463) used by leaderboard, especially in gsm8k as we observed.

Great, thanks for the clarification!

**Comment by acherstyx** [comment]
Hello @guozhiyao, I noticed that the task name and evaluation metric in version 4.0 of lm-evaluation-harness are different from those described on the [Open LLM Leaderboard](https://huggingface.co/spaces/HuggingFaceH4/open_llm_leaderboard) (About -> REPRODUCIBILITY). Could you please provide your evaluation scripts for running evaluation locally? Thanks!

**Comment by acherstyx** [comment]
Hello @yihedeng9 , I noticed that the task name and evaluation metric in version 4.0 of lm-evaluation-harness are different from those described on the [Open LLM Leaderboard](https://huggingface.co/spaces/HuggingFaceH4/open_llm_leaderboard) (About -> REPRODUCIBILITY). Could you please provide your evaluation scripts for running evaluation locally? Thanks!
